src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameOfT.cs

```diff
@@ -144,6 +135,8 @@ public class Frame<TContext> : Frame
             }
             finally
             {
+                // Error occurred, do not return components to pool
+                ResetComponents(poolingPermitted: false);
```

Comment:
Use numerics now for pool sizes, when zero it doesn't pool; by default is zero.

Only doesn't return to pool on error.